tests/check-strings/check-strings

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python3
+
+# This script checks for strings like paths or class names that are *not* in the source code, but e.g. in
+# translation files, stylesheets or git files.
+# Invalid strings *in the source code* are usually recognized when you compile them, but other strings may
+# be overseen, which is why this script checks strings *outside of the source code*.
+
+
+import os.path
+import re
+import subprocess
+import xml.etree.ElementTree as ElementTree
+from pathlib import Path
+import tinycss2
+
+
+# global variables
+
+errors = 0
+
+
+# functions
+
+def caption(my_str):
+    print('')
+    print('# ' + my_str)
+    print('')
+
+def error(where, my_str):
+    global errors
+    errors += 1
+    print('Error: ' + where + ': ' + my_str)
+
+
+# prepare some variables
+
+if not os.path.exists(".gitmodules"):
+    print('You need to call this script from the LMMS top directory')
+    exit(1)
+
+result = subprocess.run(['git', 'ls-files', 'src/', 'include/', 'plugins/'], capture_output=True, text=True, check=True)
+files = result.stdout.splitlines()
+classes = set()
+class_pat = re.compile(r'class(\s+LMMS_EXPORT)?\s+([a-zA-Z0-9_]+)')
+for cur_file in files:
+    if os.path.isfile(cur_file):
+        with open(cur_file, errors='replace') as fp:
+            lines = fp.readlines()
+            for line in lines:
+                if res := class_pat.search(line):
+                    classes.add(res.group(2))
+
+
+# the real checks
+
+caption('.gitmodules')
+
+with open(".gitmodules") as fp:
+    lines = fp.readlines()
+    for line in lines:
+        if res := re.match(r'\[submodule "([^"]+)"\]\s*$', line):
+            if not os.path.isdir(res.group(1)):
+                error('.gitmodules', 'Directory does not exist: ' + res.group(1))
+
+
+caption('locale')
+
+filenames = set()
+classes_found = set()
+for cur_file in Path('data/locale').glob('*.ts'):
+    tree = ElementTree.parse(str(cur_file))
+    root = tree.getroot()
+    for location in root.findall('./context/message/location'):
+        filenames.add(location.attrib['filename'])
+    for location in root.findall('./context/name'):
+        classes_found.add(location.text)
+for f in sorted(filenames):
+    if not os.path.isfile(f) and not os.path.isfile("data/locale/" + f):
+        error('data/locale', 'Source file does not exist: ' + f)
+for c in sorted(classes_found):
+    if c[0] != 'Q' and not '::' in c and not c in classes:
```

Comment:
Why don't you pull out the checks for classes which are excluded in this check? The condition may change later and is used in one more place.

Reply:
At least, for this case, I could remove `c[0] != 'Q'` completely. However, I can write a function for the other one.